### PR TITLE
Feature/25 enable django accounts and create login page

### PR DESCRIPTION
**Pull request by BenDubi**
Added user accounts, this included: 

- Creating registration and login pages
- Adding user authentication to access submissions and other features
- Creating a home page

### CODE REVIEW

File: src/webserver/webserver/authentication/templates/registration/login.html [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    <div class="login-container">
+      <h2>Login</h2>
+
+      {% if form.errors %}
+        <div class="error-message">
+          <p>Your username and password didn't match. Please try again.</p>
+        </div>
+      {% endif %}
+
+      {% if next %}
```
Comment (NateUy) [inline src/webserver/webserver/authentication/templates/registration/login.html:205]: What is this if-else chain for?

Comment (BenDubi) [reply src/webserver/webserver/authentication/templates/registration/login.html:205]: The first if checks if there was an error during the login, mainly if the username and password do not match any credentials in the system. The second if next is triggered when a user tries to access a page that requires being logged in or requires special privileges and sends a message to the user that they cannot access it

File: src/webserver/webserver/authentication/apps.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from django.apps import AppConfig
+
+
+class AuthenticationConfig(AppConfig):
+    default_auto_field = 'django.db.models.BigAutoField'
+    name = 'authentication'
+
+    # for dev purposes only. look into
```
Comment (AidanMariglia) [inline src/webserver/webserver/authentication/apps.py:8]: we can get rid of this comment

**Review by AidanMariglia — CHANGES_REQUESTED** [review]
Did you commit everything? When i pull it and try to run the code i get `TemplateDoesNotExist at /authentication/register` trying to navigate to the sign up page

**Comment by AidanMariglia** [comment]
> Did you commit everything? When i pull it and try to run the code i get `TemplateDoesNotExist at /authentication/register` trying to navigate to the sign up page

it looks like the template is there, will look into why this isn't working

**Review by AidanMariglia — CHANGES_REQUESTED** [review]
found the problem

**Review by AidanMariglia — APPROVED** [review]
LGTM :)

**Pull request merged**